src/renderer/dx/DxRenderer.cpp

```diff
@@ -1202,25 +1245,29 @@ enum class CursorPaintType
 // - S_OK or relevant DirectX error
 [[nodiscard]] HRESULT DxEngine::UpdateFont(const FontInfoDesired& pfiFontInfoDesired, FontInfo& fiFontInfo) noexcept
 {
-    const auto hr = _GetProposedFont(pfiFontInfoDesired,
-                                     fiFontInfo,
-                                     _dpi,
-                                     _dwriteTextFormat,
-                                     _dwriteTextAnalyzer,
-                                     _dwriteFontFace);
+    RETURN_IF_FAILED(_GetProposedFont(pfiFontInfoDesired,
+                                      fiFontInfo,
+                                      _dpi,
+                                      _dwriteTextFormat,
+                                      _dwriteTextAnalyzer,
+                                      _dwriteFontFace));
 
-    const auto size = fiFontInfo.GetSize();
+    try
+    {
+        const auto size = fiFontInfo.GetSize();
 
-    _glyphCell.cx = size.X;
-    _glyphCell.cy = size.Y;
+        _glyphCell.cx = size.X;
+        _glyphCell.cy = size.Y;
+    }
+    CATCH_RETURN();
 
-    return hr;
+    return S_OK;
 }
 
 [[nodiscard]] Viewport DxEngine::GetViewportInCharacters(const Viewport& viewInPixels) noexcept
 {
-    short widthInChars = static_cast<short>(viewInPixels.Width() / _glyphCell.cx);
-    short heightInChars = static_cast<short>(viewInPixels.Height() / _glyphCell.cy);
+    const short widthInChars = gsl::narrow<short>(viewInPixels.Width() / _glyphCell.cx);

```

Comment:
@miniksa gel::narrow_cast since this is noexcept? Or is this considered fatal if if fails?

Reply:
Oh wait we are doing division. :)

Reply:
Yep. Fixed.